kubeshop/kusk-gateway
# TLS: with static certificates

This PR closes #33 

## Changes
Adds support for specifying TLS options in an EnvoyFleet
- an array of CipherSuites
- TLS min and max versions
- an array of cert and key pairs

## Testing
- `make create-env`
- Wait for everything to become healthy (ensure cert-manager is installed)
- Apply `issuer.yaml` and `certificate.yaml` (below)
- Uncomment TLS and tlsSecrets options in the default EnvoyFleet. Leave cipherSuites and min/max versions commented out. 
- Reapply `kubectl apply -f config/samples/gateway_v1_envoyfleet.yaml`
- Install the [TODOMVC application](https://github.com/kubeshop/kusk-gateway/blob/main/docs/todomvc.md)
- Hit the application on HTTPS

### issuer.yaml
```
kubectl apply -f - <<EOF
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: test-selfsigned
  namespace: default
spec:
  selfSigned: {}
EOF
```

### certificate.yaml
```
kubectl apply -f - <<EOF
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: my-cert
  namespace: default
spec:
  dnsNames:
    - todomvc.com
  secretName: my-cert
  issuerRef:
    name: test-selfsigned
EOF
```

## Checklist
